plugins/catalog-backend/src/ingestion/processors/BuiltinKindsEntityProcessor.ts

```diff
@@ -259,6 +285,8 @@ export class BuiltinKindsEntityProcessor implements CatalogProcessor {
         RELATION_HAS_MEMBER,
         RELATION_MEMBER_OF,
       );
+
+      // TODO: Create an attachment for the photo by downloading the URL.
```

Comment:
The photo use case is quite interesting. Here the choice for the location format is quite pretty. Providers like GitHub provide the Photo as an URL you can access without auth. There we could just use the `url:` location type instead of duplicating the data into the catalog. Other providers like Microsoft Graph require that you copy the data as they don't provide a public URL.

Reply:
Yeah it could be possible to transparently handle those things if the end consumer calls stuff like `url()` or `blob()`. Then we can got fetch data when needed, or create an object URL when needed. Would expect this to be forwards compatible with that though, so could prolly wait

Reply:
So if we want to support `getAttachment().blob()` transparently for these case, we should rethink some things.

The current implementation is:

```
...
spec:
  # We reference a location, which we resolve to be located inside the attachment storage under name myname.
  # catalogClient.getAttachment(entity, 'myname').blob()
  attachmentXLocation: attachment:myname
```
```
...
spec:
  # We reference a location, of type url so we can do a simple fetch from example.com/myname
  # so something like fetch(entity.spec.attachmentXLocation.target)
  attachmentXLocation: url:example.com/myname
```

This isn't transparent to the caller at all. He has to take care which location type is used and has to use the right API afterwards. I don't have a good idea how to solve this in the current design - One idea would be to pass the location into `catalogClient.getAttachment()` instead of the attachment name, like `catalogClient.getAttachment(entity, entity.spec.attachmentXLocation)`.

Another alternativ is to get rid of the indirection inside the entity format and directly call the attachment API, like `catalogClient.getAttachment(entity, 'attachment-x')`. This makes it harder to discover which attachments are available inside an entity kind - but that is something we could solve in the documentation. However, this also removes static checking that `attachment-x` is filled, as we have with the schemas now. We would have to change the storage in the backend to support both data and url.

While this design is open to other location types, it's hard to support other than `url:` and `attachment:`. The requirement for `url:` is, that it is accessible from the scope of the user (e.g. correct network, correct auth, ...). Maybe it would be easier to not support the `url:` case and to always resolve them in the backend using the `UrlReader` and storing them in the backend - Always storing them would also be more resilient and independent from other systems - with the trade off of storing more data in the catalog and not having real time updates.

I'm not really sure which way to go 🤷‍♂️